cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -635,19 +636,19 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) caerrors.AutoscalerErr
 			scaleDownStatus.RemovedNodeGroups = removedNodeGroups
 		}
 
+		scaleDownStatus = a.scaleDownActuator.ScaleDownStatus()
```

Comment:
Why not do this where `scaleDownStatus` is defined?

Reply:
Good question, currently I'm trying to keep it as close to original place of capturing in order to get 100% same behavior and state of data as was intended before my changes. If you are saying that it's safe to move it to status initialization - I may do that

Reply:
IMO the best place to set the results is actually in the defer block, right before the processor is called. It's consistent with how `UnremovableNodesInfo` is set, and it makes the most sense to me semantically - "before calling the processor, finalize the status by filling all asynchronously-obtained information".

There shouldn't be any change in behavior - we care about reflecting the results in the status as soon as the tracker reports them, and only reflecting each result once. IIUC before and after the changes, each result should only be reflected in the status once, since we always clear the results immediately after reflecting. After the changes, if anything the results will be reported earlier than before the changes (since the defer call will always happen regardless of the rest of the loop).

Reply:
Thanks for explanations! Moved node deletion results collection to defer statement